### PR TITLE
Overhaul the primary READMEs

### PR DESCRIPTION
**Pull request by stuartmorgan**
Streamline and simplify the READMEs:
- Moves the setup and `flutter run` documentation to the main page,
  condensing the setup description slightly.
- Moves the IDE setup to the main page; it's now much shorter, and
  slightly less step-by-step, but having it on the main page makes it
  much more discoverable.
- Adds a bit of discussion of how example/ is different from a stock
  template app to make the changes more discoverable.
- Eliminates Quick-Start, moving that content to the example README.

This makes the flow much simpler: the main README has all the
information to get started, including the command to actually run the
example. Then the example README has more information about how it works
and how it can be re-used.

Improving the plugin READMEs is left for a follow-up.

### CODE REVIEW

**Comment by stuartmorgan** [comment]
@csells This is the pending overhaul discussed last week.

**Comment by csells** [comment]
LGTM. My questions all boil down to how I think about this README vs. this wiki entry: https://github.com/flutter/flutter/wiki/Desktop-shells

File: README.md [hunk 1]
```diff
@@ -1,65 +1,95 @@
...
 This project was originally created to develop Windows, macOS, and Linux
-implementations of [Flutter](https://github.com/flutter/flutter). That work has
-since become part of the
-[Flutter engine repository](https://github.com/flutter/engine), and this
+embeddings of [Flutter](https://github.com/flutter/flutter). That work has
+since [become part of
+Flutter](https://github.com/flutter/flutter/wiki/Desktop-shells), and this
 project is now just an example of, and test environment for, building
-applications using those libraries.
+applications using those libraries in their current state.
```
Comment (csells) [inline README.md]: Would be great to also mention the experimental plugins.

File: README.md [hunk 2]
```diff
@@ -1,65 +1,95 @@
...
-desktop support](https://github.com/flutter/flutter/wiki/Desktop-shells).
+As explained in the link above, desktop libraries are still in early stages.
+**The code here is not stable, nor intended for production use.**
 
-## How to Use This Code
+## Setting Up
 
-_If you have an existing Flutter app and just want to get it running, see
-the [quick start](Quick-Start.md) page before continuing._
+Everything in this project requires [enabling desktop support in
```
Comment (csells) [inline README.md]: It's confusing to me to see a link to this same info on another doc that says essentially the same thing. I'm not sure why I'm being redirected to another page in this case.

File: README.md (hunk 1 above)
Comment (stuartmorgan) [reply README.md]: Done.

File: README.md (hunk 2 above)
Comment (stuartmorgan) [reply README.md]: I link to the other page to provide context for why these steps are necessary.

This version includes more step-by-step details, such as the exact commands to set the environment variable.

**Pull request merged**